skbuild/platform_specifics/windows.py

```diff
@@ -359,6 +362,7 @@ def _get_msvc_compiler_env(vs_version, vs_toolset=None):
             out = subprocess.check_output(
                 'cmd /u /c "{}" {} {} && set'.format(vcvarsall, arch, vcvars_ver),
                 stderr=subprocess.STDOUT,
+                shell=True,
```

Comment:
```suggestion
                shell=sys.platform.startswith("Cygwin"),
```

🤦

Reply:
```pycon
>>> sys.platform
'cygwin'
```
if you decide to go that route.

Reply:
My system (Windows 10) shows sys.platform as "cygwin" in lowercase.

In general, the PR should also document in the readme or somewhere to install python using the Cygwin installer to prevent Cygwin from running the Windows version (otherwise Cygwin uses the Windows version of Python if it is in the PATH (where `sys.platform == "win32"` and `platform.system()=="Windows"` and `platform.platform()==#something like "Windows-10-10.0.19041-SP0"`).

Reply:
It is recommended and [idiomatic](https://docs.python.org/3/library/sys.html#sys.platform) to check `sys.platform` with `startswith`, it used to matter on some systems. I don't think it ever mattered with `cygwin`, but it did on some other platforms (they might be all simpler names since Python 3.8 now, I don't know if they are all covered).

> otherwise Cygwin uses the Windows version of Python

In that case, wouldn't it just build and behave like normal Windows?

Reply:
Subprocess is a little weird on Windows (we have to have special workarounds for bugs in cibuildwheel). I'd be okay to go with the above workaround for now, since it doesn't change the current Windows behavior (though I don't understand why the current behavior works...)